src/fast_agent/agents/tool_agent.py

```diff
@@ -96,18 +95,23 @@ async def generate_impl(
             )
 
             if LlmStopReason.TOOL_USE == result.stop_reason:
-                self._tool_loop_error = None
+                tool_message = await self.run_tools(result)
+                error_channel_messages = (tool_message.channels or {}).get(FAST_AGENT_ERROR_CHANNEL)
+                if error_channel_messages:
+                    tool_result_contents = [
+                        content
+                        for tool_result in (tool_message.tool_results or {}).values()
+                        for content in tool_result.content
+                    ]
+                    if tool_result_contents:
+                        if result.content is None:
```

Comment:
By Looking at [this](https://github.com/evalstate/fast-agent/blob/main/src/fast_agent/mcp/prompt_message_extended.py#L27), I imagine this is line is unnecessary, right?